js/util/mapbox.js

```diff
@@ -80,18 +106,21 @@ exports.normalizeTileURL = function(tileURL, sourceURL, tileSize) {
     return formatUrl(urlObject);
 };
 
-function replaceTempAccessToken(params) {
+function replaceTempAccessToken(params: Array<string>) {
     for (let i = 0; i < params.length; i++) {
         if (params[i].indexOf('access_token=tk.') === 0) {
-            params[i] = `access_token=${config.ACCESS_TOKEN}`;
+            params[i] = `access_token=${config.ACCESS_TOKEN || ''}`;
         }
     }
 }
 
 const urlRe = /^(\w+):\/\/([^/?]+)(\/[^?]+)?\??(.+)?/;
 
-function parseUrl(url) {
+function parseUrl(url: string): ?UrlObject {
     const parts = url.match(urlRe);
+    if (!parts) {
+        return null;
+    }
```

Comment:
Can we avoid all the repetitive null checks and ambiguity here by just throwing an error?